invenio_rdm_records/services/access/service.py

```diff
@@ -365,6 +365,13 @@ def create_grant(self, identity, id_, data, expand=False, uow=None):
             data, context={"identity": identity}, raise_errors=True
         )
 
+        for grant in parent.access.grants:
+            if (
+                grant.subject_id == data["subject"]["id"]
+                and grant.subject_type == data["subject"]["type"]
+            ):
+                raise GrantExistsError()
+
```

Comment:
Prevents from creating 2 grants for the same user/role for the same record

Reply:
Please test in the UI what error will be displayed, making sure that it is understandable ;)

Reply:
Of course, I will test it when we have the UI for this 😃